apps/management/lib/management/view_state.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Management.ViewState do
+  @moduledoc """
+  View state management behaviour and default implementation on a per-collection
+  basis. `ViewState` collections should track only one type of object.
+
+  ## Configuration
+
+  Configure `ViewState` with `use/2` macro.
+
+  * `instance` - Brook instance name. See `Brook.instance()` for more info.
+  * `collection` - Name for object grouping. See `Brook.view_collection()` for more info.
+  """
+
+  @callback collection() :: Brook.view_collection()
+  @callback persist(Brook.view_key(), map) :: :ok
+  @callback get(Brook.view_key()) :: {:ok, map} | {:error, term}
+  @callback get_all() :: {:ok, [map]} | {:error, term}
+  @callback delete(Brook.view_key()) :: :ok
+
+  defmacro __using__(opts) do
+    instance = Keyword.fetch!(opts, :instance)
+    collection = Keyword.fetch!(opts, :collection)
+
+    quote location: :keep do
+      @behaviour Management.ViewState
+
+      @impl true
+      def collection do
+        unquote(collection)
+      end
+
+      @impl true
+      def persist(key, object) do
+        unquote(collection)
+        |> Brook.ViewState.merge(key, object)
+      end
+
+      @impl true
+      def get(key) do
+        unquote(instance)
+        |> Brook.get!(unquote(collection), key)
+        |> Ok.ok()
+      catch
+        _, reason -> {:error, reason}
+      end
+
+      @impl true
+      def get_all do
+        unquote(instance)
+        |> Brook.get_all(unquote(collection))
```

Comment:
I'm still undecided if I like this returning the entire key/value view state or if I want to call `Brook.get_all_values/2` here.

Reply:
I would do the latter. That's generally what I've found myself wanting when I call `get_all`.

Reply:
get all values seems limiting but maybe we never find that use case where we need to unwrap the keys.

Reply:
I'm not going to change it in this PR, but I'm going to play with changing it in the next PR.